src/services/redisSessionStorage.ts

```diff
@@ -171,6 +173,24 @@ export default class RedisSessionStorage extends RedisStorageUtils
       );
     });
 
+    const setFIMSToken = new Promise<Either<Error, boolean>>(resolve => {
```

Comment:
Maybe its time to think about a refactor to extract the common code. Every `setXXXXToken` is equal to the others.

Reply:
You are right, but I prefer doing the refactor on another PR with only the removal of duplicated code.